core/internal/filetransfer/file_transfers.go

```diff
@@ -19,7 +21,16 @@ type ReferenceArtifactFileTransfer interface {
 type FileTransfers struct {
 	// Default makes an HTTP request to the destination URL with the file contents.
 	Default FileTransfer
+
+	// GCS connects to GCloud to upload/download files given their paths
 	GCS     ReferenceArtifactFileTransfer
+	GCSOnce *sync.Once
```

Comment:
does it make sense to move this inside the GCSFileTransfer?

Reply:
yeah that makes sense to me, we can just attempt to set up the client first thing in the Download/Upload functions instead